sdk/src/main/java/io/radar/sdk/RadarGoogleLocationClient.kt

```diff
@@ -0,0 +1,130 @@
+package io.radar.sdk
+
+import android.annotation.SuppressLint
+import android.app.PendingIntent
+import android.content.Context
+import android.location.Location
+import com.google.android.gms.location.*
+
+@SuppressLint("MissingPermission")
+internal class RadarGoogleLocationClient(
+    context: Context,
+    private val logger: RadarLogger
+): RadarAbstractLocationClient() {
+
+    @SuppressLint("VisibleForTests")
+    val locationClient = FusedLocationProviderClient(context)
+    val geofencingClient = GeofencingClient(context)
+
+    override fun getCurrentLocation(desiredAccuracy: RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy, block: (location: Location?) -> Unit) {
+        val priority = when(desiredAccuracy) {
+            RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.HIGH -> LocationRequest.PRIORITY_HIGH_ACCURACY
+            RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.MEDIUM -> LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY
+            RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.LOW -> LocationRequest.PRIORITY_LOW_POWER
+            RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy.NONE -> LocationRequest.PRIORITY_NO_POWER
+        }
+
+        logger.d("Requesting location")
+
+        locationClient.getCurrentLocation(priority, null).addOnSuccessListener { location ->
+            block(location)
+        }.addOnCanceledListener {
+            block(null)
+        }
+    }
+
+    override fun requestLocationUpdates(
+        desiredAccuracy: RadarTrackingOptions.RadarTrackingOptionsDesiredAccuracy,
+        interval: Int,
+        fastestInterval: Int,
+        pendingIntent: PendingIntent
+    ) {
+        val priority = when(desiredAccuracy) {
```

Comment:
move this priority translation to a method, since we're repeating it above, too?

Reply:
Fixed